Automattic/newspack-plugin
# Performance: optimize analytics

### All Submissions:






### Changes proposed in this Pull Request:



- refactors the `class-analytics.php` to three separate files (general, events handling, dimensions handling)
- changes the analytics (non-AMP) JS code sent to the front-end to an optimised version with minimal repetitions and removed whitespace  

Closes 1202855962089497-as-1203447234567381

### How to test the changes in this Pull Request:

1. Ensure the site has AMP plugin off
2. Ensure you've GA connected via Site Kit* and NTG events switched on in in Newspack Analytics wizard
3. Ensure all kinds of events will fire:
    1. For click events, insert this HTML in a tested page (simulates a sharing button): `<a href="http://twitter.com" target="_blank" rel="noreferrer noopener" class="share-twitter">Twitter!</a>`
    2. Make sure at least one overlay prompt will be displayed on the tested page 
4. Observe all kinds of events are firing**:
    1. Prompt `"Load"` events on page load
    2. Prompt `"Seen"` events when prompts are scrolled into view
    3. NTG "article milestone" events on scrolling a post
    4. NTG click event on clicking the Twitter link 
    5. Form-submission-triggered `"Dismissal"` events when a prompt is dismissed
5. Inspect the page source on `master` and on this branch and observe the difference in the size of analytics-related HTML (the `script` tags)

\* on local dev. environment, you'll need the following filter to make Site Kit send data to GA:

```
add_filter(
	'googlesitekit_allowed_tag_environment_types',
	function( $types ) {
		$types[] = 'local';
		return $types;
	}
);
```

\** Filter the network tab by `collect` and inspect the payloads. Note that if GA4 is configured, events will be sent to both GA and GA4. GA4 may batch events sent at similar time. 

### Other information:
